scripts/booster-integration-tests.sh

```diff
@@ -53,12 +53,16 @@ for dir in $(ls -d test-*); do
             SERVER=$BOOSTER_DEV ./runDirectoryTest.sh test-$name $@
             ;;
         "log-simplify-json")
-            SERVER=$KORE_RPC_BOOSTER SERVER_OPTS="--log-format json -l Simplify --log-file test-$name/simplify-log.txt" ./runDirectoryTest.sh test-$name $@
+            SERVER=$KORE_RPC_BOOSTER SERVER_OPTS="--post-exec-simplify --fallback-simplify --log-format json -l Simplify --log-file test-$name/simplify-log.txt" ./runDirectoryTest.sh test-$name $@
             ;;
         "foundry-bug-report")
             SERVER=$KORE_RPC_BOOSTER ./runDirectoryTest.sh test-$name --time $@
             SERVER=$KORE_RPC_BOOSTER SERVER_OPTS="--interim-simplification 100" ./runDirectoryTest.sh test-$name $@
             ;;
+        "issue3764-vacuous-branch")
+            # must discover the vacuous branch during post-exec simplification
+            SERVER=$KORE_RPC_BOOSTER SERVER_OPTS="--post-exec-simplify" ./runDirectoryTest.sh test-$name $@
+            ;;
```

Comment:
This test was specifically addressing the problem that a vacuous branch was only recognised as such in the post-exec simplification (see issue #3764).

Reply:
I think we are unlikely to end-up in this situation in the first place, because Booster checks consistency of ensured conditions and would report if it makes the pattern `#Bottom` during rewriting, i.e. executing the state
  ```
  Term:
      <generatedTop>(
          <k>(kseq("init", dotk())),
          <int>("0"),
          <generatedCounter>("0")
      )
  Conditions:
  ```

  results in a vacuous response after an attempt to apply the `init` rule:
  ```
  "result": {"reason":"vacuous","depth":0, ...
  ```

  and the logs would show:

  ```
  [booster][execute][term e7bdd85][rewrite 29b5f78][detail] TEST.init
  [booster][execute][term e7bdd85][rewrite 29b5f78][match][success] TEST.init
  [booster][execute][term e7bdd85][rewrite 29b5f78][match][success][substitution] Substitution: Rule#Var_DotVar0:SortGeneratedCounterCell{} -> <generatedCounter>("0") , Rule#VarX:SortInt{} -> "0" , Rule#Var_Gen4:SortIntCell{} -> <int>("0") , Rule#Var_DotVar1:SortK{} -> dotk()
  [booster][execute][term e7bdd85][rewrite 29b5f78][constraint][term d6e7b73][kore-term] _==Int_("0", "42")
  [booster][execute][term e7bdd85][rewrite 29b5f78][constraint][term d6e7b73][hook INT.eq][success][term 9eb81e8][kore-term] "false"
  [booster][execute][term e7bdd85][rewrite e7fa1c1][detail] TEST.JNIT-E
  [booster][execute][term e7bdd85][rewrite e7fa1c1][match][failure] Values differ:"jnit" "init"
  [booster][execute][term e7bdd85][rewrite 928fbc2][detail] TEST.JNIT-F
  [booster][execute][term e7bdd85][rewrite 928fbc2][match][failure] Values differ:"jnit" "init"
  [booster][execute][term e7bdd85][kore-term] <generatedTop>(<k>(kseq("init", dotk())), <int>("0"), <generatedCounter>("0"))
  [booster][execute][term e7bdd85] Simplified to bottom after 0 steps.
  ```

We are fine in `"execute"` requests, because Booster would never produce `state-vacuous-but-rewritten.execute` anymore. It would have before, as we did not have the SMT solver to prune the contradictory ensured conditions.

I have a related comment about the `"simplify"` requests, but I'll post it in a separate comment.